### ISSUE TITLE
Relax SBOM validation

### ISSUE THREAD

**Issue by dmikusa-pivotal** [buildpacks/libcnb#126]
We [validate SBOM formats in libcnb](https://github.com/buildpacks/libcnb/blob/main/build.go#L393), but this only takes into consideration the core mime type, not any additional parameters/specifiers.

We need to improve the validation to take into consideration additional parameters or relax it so that they are ignored.

**Comment by sagnik3788** [comment]
@dmikusa , may I give it a try?

**Comment by dmikusa** [comment]
Absolutely. Let me know if you have questions or need me to take a look at a PR.

**Comment by sagnik3788** [comment]
What additional parameters can I add to `validateSBOMFormats` Can I include a `CustomValidator` or `additionalMetadata` if required? @dmikusa

**Comment by dmikusa** [comment]
@sagnik3788 Sorry for the delay, I was at a conference last week.

Technically you can add any parameters you want. That's a private method so we don't need to worry about breaking changes there.

My suggestion would be that you start by using the `mime` library to parse the mime types. One issue with the current implementation is that we process them as strings, which works Ok for basic mime types like `application/zip` but more complicated mime types like `text/plain;charset=us-ascii` will break it. Using the mime library will process those correctly.

The link above has an example of that.

**Comment by sagnik3788** [comment]
@dmikusa can you review the pr

**Comment by loewenstein** [comment]
@dmikusa  is this in the 2.0 milestone because it cannot be done in 1.x or because it is a requirement to release 2.0?
